training/deepspeech_training/util/helpers.py

```diff
@@ -78,15 +81,44 @@ def __len__(self):
         return self.len
 
 
+class LenMap:
+    """
+    Wrapper around python map() output object that preserves the original collection length
+    by implementing __len__.
+    """
+
+    def __init__(self, fn, iterable):
+        try:
+            self.length = len(iterable)
+        except TypeError:
+            self.length = None
+        self.mapobj = map(fn, iterable)
+
+    def __iter__(self):
+        self.mapobj = self.mapobj.__iter__()
+        return self
+
+    def __next__(self):
+        return self.mapobj.__next__()
+
+    def __getitem__(self, key):
+        return self.mapobj.__getitem__(key)
+
+    def __len__(self):
+        return self.length
+
+
 class LimitingPool:
     """Limits unbound ahead-processing of multiprocessing.Pool's imap method
     before items get consumed by the iteration caller.
     This prevents OOM issues in situations where items represent larger memory allocations."""
+
     def __init__(self, processes=None, initializer=None, initargs=None, process_ahead=None, sleeping_for=0.1):
         self.process_ahead = os.cpu_count() if process_ahead is None else process_ahead
         self.sleeping_for = sleeping_for
         self.processed = 0
-        self.pool = Pool(processes=processes, initializer=initializer, initargs=initargs)
+        self.pool = Pool(processes=processes,
+                         initializer=initializer, initargs=initargs)
```

Comment:
OK I see there's a bunch of formatting changes in this file. Please remove them as it makes `git blame` harder to read. What we should do instead is an atomic pass of `black` on the whole code at once and then start using it from that point on.

Reply:
Yeah! OK! I'll revert!